Source/ASRunLoopQueue.mm

```diff
@@ -147,37 +165,73 @@ - (void)stop
   _thread = nil;
 }
 
-- (void)test_drain
+- (void)_stop
 {
-  [self performSelector:@selector(_test_drain) onThread:_thread withObject:nil waitUntilDone:YES];
+  CFRunLoopStop(CFRunLoopGetCurrent());
 }
 
-- (void)_test_drain
+- (void)dealloc
 {
-  while (true) {
-    @autoreleasepool {
-      _queueLock.lock();
-      std::deque<id> currentQueue = _queue;
-      _queue = std::deque<id>();
-      _queueLock.unlock();
+  [self stop];
+}
 
-      if (currentQueue.empty()) {
-        return;
-      } else {
-        currentQueue.clear();
-      }
-    }
-  }
```

Comment:
V1 is untouched except to remove the unused `test_drain` and `_test_drain` methods. The diff is hard to read in unified view =/